tests/users/test_api_resources.py

```diff
@@ -137,6 +135,161 @@ def test_user_registration_mentee(self, send_email_function):
+    def test_user_registration_with_invalid_length_username(self, send_email_function):
+        with self.client:
+            # invalid username and password length
+            response = self.client.post(
+                "/register",
+                data=json.dumps(
+                    dict(
+                        name=user1["name"],
+                        username="test",
+                        password=user1["email"],
+                        email=user1["email"],
+                        terms_and_conditions_checked=user1[
+                            "terms_and_conditions_checked"
+                        ],
+                    )
+                ),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+
+            user = UserModel.query.filter_by(username="test").first()
+            self.assertIsNone(user)
+            self.assertEqual(response.status_code, 400)
+
+    @patch("flask_mail._MailMixin.send", side_effect=mail_send_mocked)
+    def test_user_registration_with_invalid_length_password(self, send_email_function):
+        with self.client:
+            # invalid username and password length
+            response = self.client.post(
+                "/register",
+                data=json.dumps(
+                    dict(
+                        name=user1["name"],
+                        username=user1["username"],
+                        password="test",
+                        email=user1["email"],
+                        terms_and_conditions_checked=user1[
+                            "terms_and_conditions_checked"
+                        ],
+                    )
+                ),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+
+            user = UserModel.query.filter_by(username=user1["username"]).first()
+            self.assertIsNone(user)
+            self.assertEqual(response.status_code, 400)
+
+    @patch("flask_mail._MailMixin.send", side_effect=mail_send_mocked)
+    def test_user_registration_with_space_in_password(self, send_email_function):
+        with self.client:
+            # invalid username and password length
+            response = self.client.post(
+                "/register",
+                data=json.dumps(
+                    dict(
+                        name=user1["name"],
+                        username=user1["username"],
+                        password="test password",
+                        email=user1["email"],
+                        terms_and_conditions_checked=user1[
+                            "terms_and_conditions_checked"
+                        ],
+                    )
+                ),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+
+            user = UserModel.query.filter_by(username=user1["username"]).first()
```

Comment:
Hi @NenadPantelic can you please assert the message returned by response too? Same for other tests too

Reply:
Yeah @devkapilbansal, that sounds reasonable :wink:

Reply:
Thanks for considering